src/sdk/module/storage.h

```diff
@@ -57,19 +62,30 @@ class StorageBucket {
 
   virtual void remove_all() = 0;
   virtual ~StorageBucket() = default;
+  virtual void destroy() {};
+
+  boost::asio::io_service::strand* get_owner_strand() { return owner_strand; };
+  void set_owner_strand(boost::asio::io_service::strand* strand) { owner_strand = strand; };
```

Comment:
a bucket shouldn't need to worry about the strand, this should be implemented in a wrapper bucket class